CONTRIBUTE.md

```diff
@@ -0,0 +1,333 @@
+
+For example:
+
+```
+var array1 = ['a', 'b', 'c'];
+
+console.log(array1);
+// expected output: Array ["a", "b", "c"]
+
+var array2 = [1, 2, 3];
+
+console.log(array2);
+// expected output: Array [1, 2, 3]
+```
+
+An alternative to using numerical suffixes could be to use more descriptive suffixes, such as `arrayLetters` and `arrayNumbers` in the above examples.
+
+This may make more sense in the context of the following example:
+
+```
+var collatorDe = new Intl.Collator('de', { usage: 'search', sensitivity: 'base' });
+var collatorFr = new Intl.Collator('fr', { usage: 'search', sensitivity: 'base' });
+```
+
+### Indicating console.log output
+
+To indicate the ouput we expect, we place a comment `// expected output:` on the line below each `console.log`.
+
+
+### Representing Browser Differences
+
+Occasionally browsers will produce different results, in these cases we can denote the varying output like so:
+
+```
+console.log(matchesDe);
+// expected output (Chrome / Firefox): Array ["Bären"]
+// expected output (Safari): Array ["Bären", "Baren"]
+```
+
+## Language Choice
+
+For more established example content (such as arrays) it is recommended that we stick to ES5, where examples are required for APIs standardised after ES6 it is felt that we can use ES6 to illustrate the examples.
+
+ES6 examples should use:
+
+- `let` and `const` instead of `var`
+- Arrow functions (`=>`) for Anonymous Functions (see below)
+- Template literals (``string text ${expression} string text``)
+- Spread syntax (`myFunction(...iterableObj);`)
+
+## Providing Context
+
+Where possible it may be useful to provide examples showing how a method could be used in the context of a function or class. It may not always be possible to provide a succinct example and a series of `console.log`s can also be a valid way of illustrating an example.
+
+## Tangible examples
+
+It's nice to be able to illustrate a programming concept using real or imaginary world examples, if possible using the same theme throughout a specific topic.
+
+For example:
+
+```
+function 1(disposition) {
+  this.disposition = disposition;
+}
+
+var handler1 = {
+  construct: function(target, args) {
+    console.log("1 constructor called");
+    // expected output: " constructor called"
+
+    return new target(...args);
+  }
+};
+
+var proxy1 = new Proxy(1, handler1);
+
+console.log(new proxy1("fierce").disposition);
+// expected output: "fierce"
+```
+
+
+## Dealing with Errors
+
+Writing code that will throw an error can be useful way to illustrate an example. However an uncaught error will halt execution immediately and will not display the results of any `console.log` statements in the example.
+
+If you wish to use an error to illustrate a method you could wrap it in a `try` `catch` clause, for example:
```

Comment:
Nit: "If you wish to use an error to illustrate a method, wrap it in a `try/catch` block for example:"

Reply:
OK - I'm trying not to be too prescriptive, but maybe the intention of this doc is to be prescriptive. So 👍 :)

Reply:
Also I just noticed you changed the 'phrasing' of `try/catch` which is fine.